deploy/examples/secure-query-with-keycloak-sidecar-values.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: v1
+kind: Jaeger
+  metadata:
+    name: jaeger
+    namespace: default
+  spec:
+    agent:
+      config: {}
+      options: {}
+      resources: {}
+    allInOne:
+      config: {}
+      options: {}
+      resources: {}
+    collector:
+      config: {}
+      options: {}
+      resources: {}
+    ingester:
+      config: {}
+      options: {}
+      resources: {}
+    ingress:
+      enabled: true
+      security: "oauth-proxy"
+    query:
+      oauthProxy:
+        options:
+          #        Client ID for OpenID server
+          - --client-id=jaeger-client
+          #        Client secret for OpenID server.
+          #        Confidential access type is for server-side clients that need to perform a browser login and require a client
+          #        secret when they turn an access code into an access token, (see Access Token Request in the OAuth 2.0 spec for
+          #        more details). This type should be used for server-side applications.
+          #        see https://www.keycloak.org/docs/6.0/server_admin/
+          - --client-secret=12345678-9abc-def0-1234-56789abcdef0
+          #        URL for OpenID autoconfiguration
+          - --discovery-url=https://keycloak.example.com/auth/realms/jaeger-realm
+          - --enable-default-deny=false
+          - --enable-json-logging=true
+          - --enable-logging=true
+          - --enable-request-id=true
+          - --enable-security-filter=true
+          - --http-only-cookie=true
+          #        The interface definition you wish the proxy to listen, all interfaces is specified as ':<port>', unix sockets as unix://<REL_PATH>|</ABS PATH>
+          - --listen=:9091
+          - --preserve-host=true
+          #        The redirection url, essentially the site url, note: /oauth/callback is added at the end
+          - --redirection-url=https://jaeger-gatekeeper.example.com
+          #        Role restrictions for URL. User must have specified role in order to access.
+          - --resources=uri=/*|roles=jaeger-user
+          - --resources=uri=/metrics|roles=metrics_role
+          #        - --resources=uri=/metrics/*|white-listed=true
+          - --skip-upstream-tls-verify=true
```

Comment:
That's definitely not a good default :-)

Reply:
Indeed, `--skip-upstream-tls-verify=true` and `--skip-openid-provider-tls-verify=true` are recommended only for development purposes, but in production, they are not the best choice.

But assuming that this is just an example and most of people will try it locally prior to deploy in production. I'd suggest something like:

1. [Let's Encrypt](https://github.com/louketo/louketo-proxy/blob/master/docs/user-guide.md#lets-encrypt-configuration)
2. Enabling self-signed certificates

```
enable-self-signed-tls: true
self-signed-tls-expiration: 87600h0m0s
self-signed-tls-hostnames:
  - localhost
```

3. Provide concrete examples and instructions for people to add real certificates.

Reply:
Indeed this settings are pasted from another sidecar we've implemented. We are currently in a development state and just played around with it. I added the suggested settings.